### ISSUE TITLE
On dev installations parameters that are set on myopenaps/preferences.json are not being copied correctly 

### ISSUE THREAD

**Issue by tzachi-dar** [openaps/oref0#1251]
**Describe the bug**
Changes that are done in myopenaps/preferences.json or insulin sensitivities on the pump don't make it to the function determine_basal().

**To Reproduce**
1) Change enableUAM on myopenaps/preferences.json.
2) Change insulin sensitivities on the pump.
(actually each of 1/2 can be done by itself).
3) Run autotune as it is run in the night. (or wait to 4.05 am)
4) Look at logs of determine_basal(). Your values have not been copied there.

A workaround to the problem can be "rm settings/autotune.json" in order to force a recreation of the file.

**Comment by tzachi-dar** [comment]
Here are my conclusions regarding the bug.
TLDR: I have noticed that ISF data is not being copied correctly from pump to code.
During the debug, I have noticed that other variables are also not being copied correctly.
The bug is in the function tuneAllTheThings More specicly in:
https://github.com/openaps/oref0/blob/86f601ea56210d7dc55babcff4a5a4e906fa9476/lib/autotune/index.js#L415 

 // reconstruct updated version of previousAutotune as autotuneOutput 
The code here starts from “autotuneOutput = previousAutotune;”
As a result variables that are on the struct will not be copied from the file profile.pump.json
Let’s look at a few variables that do not work as expected:
enableUAM
Max_iob
Actually all fields that are directly on the strurct.
ISF data, also copied from the previousAutotune and not from profile.pump.json.
That is we have:
var isfProfile = previousAutotune.isfProfile;
…
autotuneOutput.isfProfile = isfProfile;
Only the first variable in the code is used.
So, we always stay with the data that was read in the first day.

**Comment by tim2000s** [comment]
This was first raised in #1093 and then again in #1101 which was merged in and appeared to resolve it. If that’s somehow got broken, we’d probably need to work out where.  As an aside, if you’re using more than one ISF, Autotune doesn’t really work properly anyway and there have been discussions about that in #326 .

**Comment by tzachi-dar** [comment]
I'll look further to the pointers you have provided tonight.

I'm aware of the problem of autotune with more than one sensitivity, but I guess parameters should still be copied correctly in that case.

**Comment by scottleibrand** [comment]
Anything specific you need help figuring out?

**Comment by tzachi-dar** [comment]
I guess that I understand the bug now. As it has a workaround, it became less urgent.
I want to finish https://github.com/NightscoutFoundation/xDrip/pull/868 first, and then I'll send my proposed fix for further discussion.

**Comment by danamlewis** [comment]
Did this one get resolved and merged into 0.7.0 already, or is this one still needing action?

**Comment by tzachi-dar** [comment]
I'm almost sure this still happens. I'll verify this in a couple of days to make sure.

**Comment by tzachi-dar** [comment]
Actually this is even worse than I thought.
I went to myopenaps/preferences.json and changed enableUAM to true.
I did what I have described in step one as a workaround, but it still seems that enableUAM remains false in many json files.
So, I had to also change it in settings/pumpprofile.json
and then run autotune, and only then it is set correctly.
